src/rendering/RenderUtil.cc

```diff
@@ -2579,16 +2582,20 @@ void RenderUtil::Init()
   this->InitRenderEnginePluginPaths();
 
   std::map<std::string, std::string> params;
-#ifdef __APPLE__
-  // TODO(srmainwaring): implement facility for overriding the default
-  //    graphics API in macOS, in which case there are restrictions on
-  //    the version of OpenGL used.
-  params["metal"] = "1";
-#else
-  if (this->dataPtr->useCurrentGLContext)
+  if (this->dataPtr->useCurrentGLContext &&
+      this->dataPtr->apiBackend != "vulkan" &&
+      this->dataPtr->apiBackend != "metal")
+  {
     params["useCurrentGLContext"] = "1";
-#endif
-
+  }
+  if (this->dataPtr->apiBackend == "vulkan")
+  {
+    params["vulkan"] = "1";
+  }
+  else if (this->dataPtr->apiBackend == "metal")
+  {
+    params["metal"] = "1";
+  }
```

Comment:
set `params["metal"] = "1"` if `this->dataPtr->apiBackend` is empty string on `APPLE`?

Reply:
If I remember correctly, no.

Because the higher level code should insist on using "metal" (and by higher level, I mean MinimalScene).

If the string is empty that means something went wrong with MinimalScene in gz-gui; and gz-gui will initialize Qt with OpenGL instead of Metal. See `GzRenderer::SetGraphicsAPI` & `RenderThread::SetGraphicsAPI`.

Qt must be initialized with Metal for this to work.

Reply:
ok I see that the gui does the right thing and set the api backend to metal. I added a change in the sensors system to do the same thing in [5d9a5fc](https://github.com/gazebosim/gz-sim/pull/1735/commits/5d9a5fc3706c3528eda9336570c13972c169896d)